src/new-components/text/text.tsx

```diff
@@ -43,6 +43,7 @@ const Text = forwardRefWithAs<TextProps>(function Text(
                 lineClampMultipleLines ? styles.lineClamp : null,
                 lineClamp ? getClassNames(styles, 'line-clamp', lineClamp.toString()) : null,
             ]}
+            paddingRight={lineClamp ? 'xsmall' : undefined}
```

Comment:
I would add a comment above this line with a link to this issue to make it clear to future readers why this prop  is added:

```suggestion
            // see https://github.com/Doist/reactist/pull/528
            paddingRight={lineClamp ? 'xsmall' : undefined}
```